test-requirements.txt

```diff
@@ -0,0 +1,14 @@
+black
+coverage[toml]
+flake8-docstrings
+flake8-builtins
+isort
+juju>=3.1
```

Comment:
Not 100% sure, but I think that the action that sets up the env for the itests will use Juju 2.9. If that's the same, integration tests in CI will fail.

Reply:
I think this is a `pythonlib-juju` version, therefore, the one used here is not a valid version.
If you mean to use `juju 3.1` then the latest `pythonlib-juju` tested with `3.1` would be `3.1.0.1` (you'd need a different version if you use `juju 2.9`)
see more [here](https://github.com/juju/python-libjuju/releases).